xtask/src/web.rs

```diff
@@ -36,22 +42,35 @@ pub fn check(sh: &Shell) -> anyhow::Result<()> {
 pub fn run(sh: &Shell) -> anyhow::Result<()> {
     let _s = Section::new("WEB-RUN");
 
-    build(sh, false)?;
-
     run_cmd_in!(sh, IRON_SVELTE_CLIENT_PATH, "{NPM} run dev-no-wasm")?;
 
     Ok(())
 }
 
-fn build(sh: &Shell, wasm_pack_dev: bool) -> anyhow::Result<()> {
+pub fn build(sh: &Shell, wasm_pack_dev: bool) -> anyhow::Result<()> {
     if wasm_pack_dev {
         run_cmd_in!(sh, IRONRDP_WEB_PATH, "wasm-pack build --dev --target web")?;
     } else {
-        let _env_guard = sh.push_env("RUSTFLAGS", "-Ctarget-feature=+simd128");
+        let _env_guard = sh.push_env("RUSTFLAGS", "-Ctarget-feature=+simd128,+bulk-memory");
         run_cmd_in!(sh, IRONRDP_WEB_PATH, "wasm-pack build --target web")?;
     }
 
-    run_cmd_in!(sh, IRON_REMOTE_GUI_PATH, "{NPM} run build-alone")?;
+    let ironrdp_web_js_file_path = sh.current_dir().join(IRONRDP_WEB_PACKAGE_JS_PATH);
+
+    let mut ironrdp_web_js_content = fs::read_to_string(&ironrdp_web_js_file_path)?;
+
+    // Modify the js file to get rid of the `URL` object.
+    // Vite doesn't work properly with inlined urls in `new URL(url, import.meta.url)`.
+    ironrdp_web_js_content = format!(
+        "import wasmUrl from './ironrdp_web_bg.wasm?url';\n\n{}",
+        ironrdp_web_js_content
+    );
+    ironrdp_web_js_content =
+        ironrdp_web_js_content.replace("new URL('ironrdp_web_bg.wasm', import.meta.url)", "wasmUrl");
+
+    fs::write(&ironrdp_web_js_file_path, ironrdp_web_js_content)?;
```

Comment:
suggestion: More idiomatic to not use `mut` here. You can shadow the previous binding.

```rust
    let ironrdp_web_js_content = fs::read_to_string(&ironrdp_web_js_file_path)?;

    // Modify the js file to get rid of the `URL` object.
    // Vite doesn't work properly with inlined urls in `new URL(url, import.meta.url)`.
    ironrdp_web_js_content = format!(
        "import wasmUrl from './ironrdp_web_bg.wasm?url';\n\n{}",
        ironrdp_web_js_content
    );
    let ironrdp_web_js_content =
        ironrdp_web_js_content.replace("new URL('ironrdp_web_bg.wasm', import.meta.url)", "wasmUrl");

    fs::write(&ironrdp_web_js_file_path, ironrdp_web_js_content)?;
```

Reply:
Done